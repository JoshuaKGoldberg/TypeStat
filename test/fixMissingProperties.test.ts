import { describe, expect, it } from "vitest";

import { checkTestResult } from "../src/tests/testSetup.js";

const cwd = import.meta.dirname;

describe("Missing properties", () => {
	it("missing property accesses", async () => {
		expect.assertions(3);
		await checkTestResult(
			cwd,
			"fixes/missingProperties/missingPropertyAccesses",
		);
	}, 10000);
});
