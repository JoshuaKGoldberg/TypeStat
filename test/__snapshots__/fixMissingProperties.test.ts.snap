// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Missing properties > missing property accesses > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": true,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;
